### DISCUSSION TITLE
Report errors "FATAL:  role "XXX" does not exist" when POSTGRES_USER is customized.

### DISCUSSION THREAD

**Discussion by Chr15t0pher** [docker-library/postgres#1155] [General]
I'm a totally newbie, trying to using  a shell script to run postgres image.
Here is my code:
```bash
DB_USER="${POSTGRES_USER:=admin}"
DB_PASSWORD="${POSTGRES_PASSWORD:=admin}"
DB_NAME="${POSTGRES_DB:=scaffold}"
DB_PORT="${POSTGRES_PORT:=5432}"
DB_HOST="${POSTGRES_HOST:=localhost}"

docker run \
  --name "postgres_$(date '+%s')" \
  -e POSTGRES_USER=${DB_USER} \
  -e POSTGRES_PASSWORD=${DB_PASSWORD} \
  -e POSTGRES_DB=${DB_NAME} \
  -p "${DB_PORT}":5432 \
  -d \
  postgres -N 1000

until PGPASSWORD="${DB_PASSWORD}" psql -h "${DB_HOST}" -U "${DB_USER}" -p "${DB_PORT}" -d "postgres" -c '\q'; do
  >&2 echo "Postgres is still unavailable - sleeping"
  sleep 1
done
```
The errors I've been able to see are:
``` bash
psql: error: connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  role "admin" does not exist
```
What am I missing and how can I fix this so I can actually proceed with Postgres?
When i set DB_USER to the default value `postgres`, it is Ok. But i just don't know the reason.

**Comment by LaurentGoderre** [comment]
I just tried this and I was able to connect. Maybe there is an issue with your docker networking. What do you get if you run `docker ps`?

**Comment by Chr15t0pher** [comment]
> I just tried this and I was able to connect. Maybe there is an issue with your docker networking. What do you get if you run `docker ps`?

I tried to run `docker ps`, it worked as i expected, i got the activated containers.
```bash
docker ps --filter 'name=postgres' --format '{{.ID}}'
```
```bash
CONTAINER ID   IMAGE      COMMAND                   CREATED         STATUS         PORTS                    NAMES
bdb2469f4a82   postgres   "docker-entrypoint.s…"   4 minutes ago   Up 4 minutes   0.0.0.0:5432->5432/tcp   postgres_1698292090
```
I thought after running
`docker run \
  --name "postgres_$(date '+%s')" \
  -e POSTGRES_USER=${DB_USER} \
  -e POSTGRES_PASSWORD=${DB_PASSWORD} \
  -e POSTGRES_DB=${DB_NAME} \
  -p "${DB_PORT}":5432 \
  -d \
  postgres -N 1000
`,  i should manually set $DB_USER. Then, i tried to modify the script like this:
```bash
docker run \
    --name "${RUNNING_POSTGRES_CONTAINER}" \
    -e POSTGRES_USER=${DB_USER} \
    -e POSTGRES_PASSWORD=${DB_PASSWORD} \
    -e POSTGRES_DB=${DB_NAME} \
    -p "${DB_PORT}":5432 \
    -d \
    postgres -N 1000

sleep 5
docker exec -it $RUNNING_POSTGRES_CONTAINER psql -U $DB_USER -d $DB_NAME -c "CREATE USER $DB_USER WITH PASSWORD '$DB_PASSWORD';"
```
However, it reports : `ERROR:  role "admin" already exists`.

**Comment by Chr15t0pher** [comment]
> I just tried this and I was able to connect. Maybe there is an issue with your docker networking. What do you get if you run `docker ps`?

According to [https://stackoverflow.com/questions/11919391/postgresql-error-fatal-role-username-does-not-exist](https://stackoverflow.com/questions/11919391/postgresql-error-fatal-role-username-does-not-exist), maybe i must use the system user `postgres` to init database?

**Comment by LaurentGoderre** [comment]
The user is successfully created. Do you have postgres installed locally as well? I am wondering if your script isn't connecting to your container postgres but another one.

**Reply by L0o0p** [reply]
you are right, thank you!, i got the similar problem, and i used"ps aux | grep postgres"to check, and i found another pgsql existed, when i removed it, theproblem was solved.

**Comment by Chr15t0pher** [comment]
> The user is successfully created. Do you have postgres installed locally as well? I am wondering if your script isn't connecting to your container postgres but another one.

Yes, i have installed locally. But i did not start local postgres, i just use the `psql` as a tool to connect to the postgres container.

**Comment by yogeshuh** [comment]
File "/home/gridsentry/imgaudit/GAudit/myenv/lib/python3.8/site-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  role "gaudit" does not exist

# Project
export ENV_HOME=$PWD
export ENV_PROJECT=GAudit
export ENV_DJANGO_SECRET_KEY='xxxxxxxxxxxxxxxxxxxxxxx
export PYTHONPATH=$ENV_HOME/backend
export ENV_SITE=im_gaudit
# Database
export ENV_DB_USER=gaudit
export ENV_DB_NAME=imdev
export ENV_DB_PASSWORD="gaudit@123"
# Super Admin
export ENV_ADMIN_USER=gsadmin
export ENV_ADMIN_PASSWORD="GS@dm!n24"
export ENV_ADMIN_EMAIL=admin@gmail.com
# Server
export ENV_SERVER_TYPE=dev
export DEBUG=1


How to fix this issue?
